### ISSUE TITLE
2005 if value is piped always wrong

### ISSUE THREAD

**Issue by kkmuffme** [koalaman/shellcheck#1438]
#### For bugs
- Rule Id (if any, e.g. SC1000): 2005
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2005)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

i="/example/example.php"
$(echo "$(basename "$i")" | sed 's/\.php//')

```

#### Here's what shellcheck currently says:
Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.

#### Here's what I wanted or expected to see:
no error. When piping something, I need echo, so it shouldn't give an error if the command is followed by a pipe

**Comment by matthewpersico** [comment]
No you don’t. Try this:

$(basename “$i” | sed ‘s/\.php//‘)

and watch what happens.

**Comment by kkmuffme** [comment]
Thanks, somehow it wasn't working when I first tried it, probably made a mistake.

**Comment by matthewpersico** [comment]
No problem.  Five bucks says you forgot the $().  I’ve been there. 😀
